### PR TITLE
[com_fields] Custom fields doesn't work with accents Fix for issue #15287

### PR DESCRIPTION
**Pull request by esedic**
Pull Request for Issue #15287.

When you create certain types of fields (tested on list and checkbox fields) and text/values have special characters like accents, you will get a warning when editing an article. See https://github.com/joomla/joomla-cms/issues/15287 for details.

This PR adds html_entity_decode() function to SimpleXMLElement string.

Pull Request for Issue # .

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 5284d050d6ae14bd092098b791d6eb1408119509<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15294">issues.joomla.org/tracker/joomla-cms/15294</a>.</sub>

**Comment by laoneo** [comment]
I would not do that in JForm itself but on the place where the load function is called. I'm not sure if the change on JForm will have some unpredictable side effects.

File: libraries/joomla/form/form.php [hunk 1]
```diff
@@ -725,7 +725,7 @@ public function load($data, $replace = true, $xpath = false)
 		{
 			try
 			{
-				$data = new SimpleXMLElement($data);
+				$data = new SimpleXMLElement(html_entity_decode($data, ENT_QUOTES, "utf-8"));
```
Comment (infograf768) [inline libraries/joomla/form/form.php]: Use single quotes

Comment (infograf768) [inline libraries/joomla/form/form.php]: and uppercase

`'UTF-8'`

**Comment by esedic** [comment]
@laoneo, I was thinking about that, if this is the right place to decode form data, but my backend knowledge is limited, I would need a little guidance here. Where is the load function called?

**Comment by laoneo** [comment]
It's in the FieldsHelper::renderForm function.

**Comment by esedic** [comment]
I can't find renderForm function in administrator/com_fields/helpers/fields.php file. Did you mean function render()?

**Comment by laoneo** [comment]
Here https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_fields/helpers/fields.php#L496

**Comment by esedic** [comment]
Ok, I hope I got it right, after making a change in helper file warning is gone. Change in JForm was reverted.

**Comment by franz-wohlkoenig** [comment]
@esedic is PR ready for Retest?

**Comment by laoneo** [comment]
Looks good for me

**Comment by Quy** [comment]
I have tested this item :red_circle: unsuccessfully on e7b203a7e2b440b7e4c70d560a16f57ee7bb0ffa<br /><br />Not this value `R&B` does not work. 
```
PHP Warning:  SimpleXMLElement::__construct(): Entity: line 2: parser error : EntityRef: expecting ';' in \joomla-cms-staging\libraries\joomla\form\form.php on line 728

PHP Warning:  SimpleXMLElement::__construct(): hint=&quot;&quot; required=&quot;false&quot; readonly=&quot;false&quot; validate=&quot;options&quot;&gt;&lt;option value=&quot;R&amp;B in \joomla-cms-staging\libraries\joomla\form\form.php on line 728

PHP Warning:  SimpleXMLElement::__construct(): ^ in \joomla-cms-staging\libraries\joomla\form\form.php on line 728

PHP Warning:  SimpleXMLElement::__construct(): Entity: line 2: parser error : EntityRef: expecting ';' in \joomla-cms-staging\libraries\joomla\form\form.php on line 728

PHP Warning:  SimpleXMLElement::__construct(): =&quot;&quot; required=&quot;false&quot; readonly=&quot;false&quot; validate=&quot;options&quot;&gt;&lt;option value=&quot;R&amp;B&quot;&gt;R&amp;B in \joomla-cms-staging\libraries\joomla\form\form.php on line 728

PHP Warning:  SimpleXMLElement::__construct(): ^ in \joomla-cms-staging\libraries\joomla\form\form.php on line 728
```
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15294">issues.joomla.org/tracker/joomla-cms/15294</a>.</sub>

**Comment by Quy** [comment]
I think I have a solution. Change `htmlentities` to `htmlspecialchars` in the following 2 lines:

https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_fields/libraries/fieldslistplugin.php#L44-L45

No more errors with the values: `A&B` and `Manuel Gómez Padilla`.

What I don't know if it requires `, ENT_COMPAT, 'UTF-8'` with `htmlspecialchars` or something else. If someone knows, please submit a pull request. Thanks.

**Comment by esedic** [comment]
Great find @Quy !
It seems that it works, I've checked it on field title, name, label and text/values for fields color, list and checkboxes.
Regarding `, ENT_COMPAT, 'UTF-8'` part: PHP 5.4 and PHP 5.5 will use UTF-8 as the default encoding. In PHP 5.6 default_charset configuration option is used as the default value. Official docs on php.net still recommends specifying charset if you are using PHP 5.5 or earlier, so probably we should include it.

**Comment by esedic** [comment]
I've updated PR, reverted previous commit to helper file and updated fieldslistplugin.php.

**Comment by laoneo** [comment]
List of changed file is empty, probably a missing commit.

**Comment by esedic** [comment]
Yep, I've made a commit to wrong branch. Done.

**Review by laoneo — APPROVED** [review]

**Comment by Quy** [comment]
I appreciate the info/pr. I am not at my computer to test, however, the pr looks good.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 99e4a645cefcfc0a50051524030d83b5c817cca0<br /><br />Created radio, checkboxes, and list custom fields. No errors on backend and frontend in article edit mode.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15294">issues.joomla.org/tracker/joomla-cms/15294</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 99e4a645cefcfc0a50051524030d83b5c817cca0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15294">issues.joomla.org/tracker/joomla-cms/15294</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by Quy** [comment]
Merge into 3.7.1?

**Pull request merged**